src/main/kotlin/org/jetbrains/research/transformations/anonimization/Anonymization.kt

```diff
@@ -0,0 +1,164 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package org.jetbrains.research.transformations.anonimization
+
+import com.github.gumtreediff.tree.ITree
+import com.github.gumtreediff.tree.TreeContext
+import org.jetbrains.research.transformations.NodeType
+import org.jetbrains.research.transformations.Transformation
+import org.jetbrains.research.transformations.nodes
+
+object Anonymization : Transformation {
+    override val metadataKey = "anonymization"
+    private const val VAR_PREFIX = "v"
+    private const val FUN_PREFIX = "f"
+    private const val ARG_PREFIX = "a"
+    private const val EMPTY_PREFIX = ""
+    private const val SEPARATOR = "_"
+
+    // Check if the node has the parent function and get its name
```

Comment:
What is a `parent function`? Could you please give an example?

Reply:
When I meet Name_Store or Arg, I want to check - which prefix should I append for it's anonymous name. For that - I look into node.parents. It's an array of parent essences. I take parent function, because I will use it's label as a prefix.

```
E.x.
def f1(a, b):
    return a + b
```

a and b have the same f1 prefix. Also, because of that - I don't nullify the counter and give them f1_a1 and f2_a2 names. When I meet another argument below (for ex in another function) - I check the prefix (it will be f2) so I make argument counter = 0 and next name will be f2_a1

Reply:
Okay, thanks :)